django/contrib/admin/static/admin/js/actions.js

```diff
@@ -113,6 +113,42 @@
             list_editable_changed = true;
         });
         $('form#changelist-form button[name="index"]').click(function(event) {
+            var actionSelected = $('select[name="action"]').val(),
+                itemSelected = $('table#result_list td.action-checkbox input:checked').length > 0,
+                messageList = $('div#container ul.messagelist'),  // admin base.html's message list
+                getMessageList = function() {
+                    return (messageList.length) ? messageList : function() {
+                        // insert in the original django admin position
+                        $('<ul class="messagelist"></ul>').insertBefore('div#content');
+                        messageList = $('div#container ul.messagelist');
+                        return messageList; // return the list (<ul><>) instance
+                    }();
+                },
+                skipChangeListFormSubmit = function(msg, type, replace) {
+                    event.preventDefault();  // skip submit
+                    if (replace) {
+                        // replace the message list with the new item error
+                        getMessageList().html('<li class="' + type + '">' + msg + '</li>');
+                    }
+                    else {
+                        // add new item error to the message list
+                        $('<li class="' + type + '">' + msg + '</li>').appendTo(getMessageList());
```

Comment:
I pushed the changes to avoid XSS issues... [d91de09](https://github.com/django/django/pull/7645/commits/d91de09d81a2867e77c434f8dcd2cf93e83ded44)